File: csharp/bob/Bob.cs
// This is a fucking dumb challenge

using System;

public static class Bob
{
    public static string Response(string statement)
    {
        // Clean this sucker up
        statement = statement.Trim();

        string msg = "Whatever.";

        if (statement.EndsWith("?"))
        {
            if (IsUpper(statement))
            {
                msg = "Calm down, I know what I'm doing!";
            }
            else {
                msg = "Sure.";
            }
        }
        else if (string.IsNullOrWhiteSpace(statement) || string.IsNullOrEmpty(statement))
        {
            msg = "Fine. Be that way!";
        }
        else if (statement.EndsWith("!") || IsUpper(statement))
        {
            msg = "Whoa, chill out!";
        }
        else if (IsUpper(statement)){
            msg = "Whoa, chill out!";
        }

        return msg;
    }

    private static bool IsUpper(string str)
    {
        bool isUpper = true;

        foreach (char s in str)
        {
            if (char.IsLetter(s) && !char.IsUpper(s)) { isUpper = false; break; }
        }

        return isUpper;
    }
}

File: python/bob/bob.py
"""
Solution to Exercism 'bob' problem.

Copied (mostly) from my Java solution.
"""


def response(hey_bob: str) -> str:
    """
    Bob is a lackadaisical teenager and responds as such.
    """

    hey_bob = hey_bob.strip()

    # First check is nothing is said
    if hey_bob == "":
        return "Fine. Be that way!"

    yell = hey_bob.isupper()
    question = False

    if hey_bob.endswith("?"):
        question = True

    if yell and question:
        return "Calm down, I know what I'm doing!"

    if yell:
        return "Whoa, chill out!"

    if question:
        return "Sure."

    # default value for any non-matching input
    return "Whatever."
